autosklearn/evaluation/train_evaluator.py

```diff
@@ -645,19 +645,31 @@ def fit_predict_and_loss(self, iterative: bool = False) -> None:
             # train_losses is a list of either scalars or dicts. If it contains dicts,
             # then train_loss is computed using the target metric (self.metric).
             if all(isinstance(elem, dict) for elem in train_losses):
-                train_loss = np.average(
-                    [
-                        train_losses[i][str(self.metric)]
-                        for i in range(self.num_cv_folds)
-                    ],
-                    weights=train_fold_weights,
-                )
+                if isinstance(self.metric, Scorer):
+                    train_loss = np.average(
+                        [
+                            train_losses[i][str(self.metric)]
+                            for i in range(self.num_cv_folds)
+                        ],
+                        weights=train_fold_weights,
+                    )
+                else:
+                    train_loss = [
```

Comment:
`train_loss` is now a list instead of just a single average number, might be okay but just checking it's still safe to do so?

Reply:
I think so. AFAIK we don't use `train_loss` anywhere.